.github/workflows/test_weaver-asset-exchange-besu.yaml

```diff
@@ -50,31 +38,44 @@ jobs:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v3.5.2
 
-      - name: Set up JDK 11
+      - uses: dorny/paths-filter@v2.11.1
+        id: changes
+        with:
+          filters: |
+            weaver_code_changed:
+              - './weaver/**!(*.md|*.css|*.html|*.jpg|*.jpeg|*.png)'
+              - '.github/workflows/test_weaver-asset-exchange-besu.yaml'
+  
+      - if: steps.changes.outputs.weaver_code_changed == 'true'
+        name: Set up JDK 11
         uses: actions/setup-java@v3.11.0
         with:
           java-version: '11'
           distribution: 'adopt'
 
-      - name: Use Node.js ${{ env.NODEJS_VERSION }}
+      - if: steps.changes.outputs.weaver_code_changed == 'true'
```

Comment:
Instead of checking at every step, what we can do is create another job say `check_code_changed` just to compute above variable `weaver_code_changed` and put it into `GITHUB_OUTPUT`,
Then for each job we just need to check only once at the start of job
```
if: needs.check_code_changed.outputs.weaver_code_changed=='true'
```
rather than at all steps, would be efficient that way I believe.

Check this workflow file for example: https://github.com/hyperledger/cacti/blob/main/.github/workflows/test_weaver-pre-release.yaml, (the subsequent jobs depend upon output of `check_release` job)